ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -19,112 +20,107 @@ export interface DropDownState {
-            <div className='argo-dropdown' ref={el => (this.el = el)}>
-                <div
-                    qe-id={this.props.qeId}
-                    className='argo-dropdown__anchor'
-                    onClick={event => {
-                        this.open();
-                        event.stopPropagation();
-                    }}>
-                    <this.props.anchor />
-                </div>
-                {ReactDOM.createPortal(
-                    <div
-                        className={classNames('argo-dropdown__content', {'opened': this.state.opened, 'is-menu': this.props.isMenu})}
-                        style={{top: this.state.top, left: this.state.left}}
-                        ref={el => (this.content = el)}>
-                        {children}
-                    </div>,
-                    document.body
-                )}
-            </div>
-        );
-    }
+export function DropDown({isMenu, anchor: Anchnor, children, qeId}: DropDownProps) {
+    const anchorRef = useRef<null | HTMLDivElement>(null);
+    const contentRef = useRef<null | HTMLDivElement>(null);
+    const subscriptionsRef = useRef<null | Subscription[]>(null);
+    const [opened, setOpened] = useState(false);
+    const [left, setLeft] = useState(0);
+    const [top, setTop] = useState(0);
 
-    public componentWillMount() {
-        this.subscriptions = [
+    useEffect(() => {
+        const content = contentRef.current;
+        const el = anchorRef.current;
+
+        subscriptionsRef.current = [
             merge(
-                dropDownOpened.pipe(filter(dropdown => dropdown !== this)),
+                dropDownOpened.pipe(filter(dropdown => dropdown !== DropDown)),
                 fromEvent(document, 'click').pipe(
                     filter((event: Event) => {
-                        return this.content && this.state.opened && !this.content.contains(event.target as Node) && !this.el.contains(event.target as Node);
+                        return content && opened && !content.contains(event.target as Node) && !el.contains(event.target as Node);
                     })
                 )
             ).subscribe(() => {
-                this.close();
+                close();
             }),
             fromEvent(document, 'scroll', {capture: true}).subscribe(() => {
-                if (this.state.opened && this.content && this.el) {
-                    this.setState(this.refreshState());
+                if (opened && content && el) {
+                    const newState = refreshState();
+
+                    setOpened(newState.opened);
+                    setLeft(newState.left);
+                    setTop(newState.top);
                 }
             })
         ];
-    }
 
-    public componentWillUnmount() {
-        (this.subscriptions || []).forEach(s => s.unsubscribe());
-        this.subscriptions = [];
-    }
+        return () => {
+            (subscriptionsRef.current || []).forEach(s => s.unsubscribe());
```

Comment:
```suggestion
            subscriptionsRef.current?.forEach(s => s.unsubscribe());
```

optional chaining simplifies this a tiny bit